apps/rosen/app/(bridge)/ChooseWalletModal.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  Grid,
+  Tooltip,
+} from '@rosen-bridge/ui-kit';
+import { Wallet, WalletInfo } from '@rosen-ui/wallet-api';
+
+interface ChooseWalletsModalProps {
+  open: boolean;
+  handleClose: () => void;
+  setSelectedWallet: ((wallet: Wallet) => Promise<void>) | undefined;
+  chainName: string;
+  supportedWallets: WalletInfo[];
+  availableWallets: Wallet[];
+}
+
+/**
+ * modal to allow user to choose a wallet
+ *
+ * @param open
+ * @param handleClose
+ * @param chainName
+ * @param supportedWallets
+ */
+export const ChooseWalletsModal = ({
+  open,
+  handleClose,
+  setSelectedWallet,
+  chainName,
+  supportedWallets,
+  availableWallets,
+}: ChooseWalletsModalProps) => {
+  const handleConnect = async (wallet: Wallet) => {
+    setSelectedWallet && (await setSelectedWallet(wallet));
+    handleClose();
+  };
+
+  return (
+    <Dialog onClose={handleClose} open={open} maxWidth="tablet">
+      <DialogContent>
+        <DialogContentText>
+          Please choose any of the supported wallets for {chainName} chain.
+        </DialogContentText>
+        <Grid container justifyContent="center" sx={{ pt: 5 }}>
+          {supportedWallets.map(
+            ({ icon: WalletIcon, label, link, name }, index) => (
+              <Tooltip title={name} key={label}>
+                <Grid
+                  item
+                  sx={{
+                    width: '8rem',
+                    height: '8rem',
+                    border: (theme) => `1px solid ${theme.palette.divider}`,
+                    borderRadius: 1,
+                    padding: 2,
+                    margin: 1,
+                    cursor: 'pointer',
+                    display: 'flex', // Added to make the Grid a flex container
```

Comment:
Please remove the comments. The code is self-explanatory. In addition, `Grid`s have flex prop themselves and all these flex props are redundant.
The comment is true for all your other changes and need a change.